File: frontend/src/services/historyService.ts
import type { HistoryRecord } from '../types/history'
import { apiRequest } from './api'

export async function getHistory(
  limit = 50,
): Promise<HistoryRecord[]> {
  return apiRequest<HistoryRecord[]>(
    `/api/history?limit=${limit}`,
  )
}

export async function searchHistory(
  query: string,
): Promise<HistoryRecord[]> {
  const normalizedQuery = query.trim()

  if (!normalizedQuery) {
    return []
  }

  return apiRequest<HistoryRecord[]>(
    `/api/history/search?query=${encodeURIComponent(normalizedQuery)}`,
  )
}

export async function getHistoryRecord(
  id: number,
): Promise<HistoryRecord | null> {
  try {
    return await apiRequest<HistoryRecord>(
      `/api/history/${id}`,
    )
  } catch {
    return null
  }
}

export async function undoHistory(
  id: number,
): Promise<boolean> {
  try {
    await apiRequest(
      `/api/history/${id}/undo`,
      {
        method: 'POST',
      },
    )

    return true
  } catch {
    return false
  }
}

File: filesense/services/history_service.py
from filesense.database import Database
from filesense.config import config

db = Database(config["db_path"])


def get_history(limit: int = 50):
    return db.get_history(limit)


def search_history(query: str):
    return db.search(query)


def get_history_record(history_id: int):
    return db.get_record(history_id)


def undo_history(history_id: int):
    return db.undo(history_id)